contracts/utils/ExternalMulticall.sol

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./interfaces/IExternalMulticall.sol";
+
+/// @notice Contract that enables calls to external contracts to be batched
+/// @dev Refer to MakerDAO's Multicall.sol for a similar implementation
+contract ExternalMulticall is IExternalMulticall {
+    /// @notice Batches calls to external contracts and reverts if at
+    /// least one of the batched calls reverts
+    /// @dev Bubbles up the raw returndata if a call reverts, leaving decoding
+    /// to the user
+    /// @param targets Array of target addresses of batched calls
+    /// @param data Array of calldata of batched calls
+    /// @param returndata Array of returndata of batched calls
+    function externalMulticall(
+        address[] calldata targets,
+        bytes[] calldata data
```

Comment:
Shouldn't we `require` that these 2 have the same `length`?

Reply:
Added